chia/full_node/mempool_manager.py

```diff
@@ -82,6 +84,69 @@ def validate_clvm_and_signature(
     return None, bytes(result), new_cache_entries
 
 
+def compute_spend_cost(
+    puzzle_reveal: SerializedProgram, solution: SerializedProgram, maximum_cost: uint64
+) -> Optional[uint64]:
+    try:
+        cost, _ = puzzle_reveal.run_with_cost(maximum_cost, solution)
+        return uint64(cost)
+    except Exception:
+        return None
+
+
+def check_item_for_dedup(
+    item: MempoolItem, dedup_coin_spends: Dict[bytes32, Tuple[bytes32, Optional[uint64]]]
```

Comment:
the `Tuple[]` at the end here, the solution hash and cost, I think it might be appropriate to actually make a small dataclass for this, to get named fields.
Also, storing a `SerializedProgram` here is cheaper than the hash, since it's just a pointer to the buffer we already have stored in a mempool item